zerver/lib/message.py

```diff
@@ -808,7 +810,7 @@ def access_web_public_message(
         raise MissingAuthenticationError
 
     try:
-        message = Message.objects.select_related().get(id=message_id)
+        message = Message.objects.select_related("recipient").get(id=message_id)
```

Comment:
What's the thinking on this being different from the list in `access_message`?

In these code paths, we are returning the object to a range of callers that want to do something with the message, and generally I'd have expected that we wanted to pass `sender` at least to all of them.

Reply:
`access_web_public_message` is only used in `json_fetch_raw_message` where no foreign keys are used. The `recipient` field is needed for `is_stream_message` call in `access_web_public_message` itself.

Reply:
I think my preference would be for us to have this match `access_message`, since I expect every place we call it will look like this, and it's way easier to reason about if both code paths out of a given block have the same properties prefetched.
```
    if not maybe_user_profile.is_authenticated:                                                         
        realm = get_valid_realm_from_request(request)                                                   
        message = access_web_public_message(realm, message_id)                                          
    else:                                                                                               
        (message, user_message) = access_message(maybe_user_profile, message_id)                        
```